### PR TITLE
corrects sniffTest behavior

### PR DESCRIPTION
**Pull request by mikebrow**
Fixes sniffTest errors on master.  Snifftest was sending kill to a sleeping detached container, additionally rm of the state directory of a container in the destroyed state does not produce the same results as runc delete.   After only deleting the state directory via rm, a subsequent runc start cid of the cid that is in the destroyed state currently fails in master, where runc delete works as desired.  

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

**Comment by mikebrow** [comment]
@duglin pls review when time allows.

File: tests/sniffTest [hunk 1]
```diff
@@ -68,15 +68,13 @@ if [[ $rc != 0 || -s err ]]; then
 	echo Output: $(cat out)
 	exit 1
 fi
-$RUNC kill c1
-rm -rf /run/opencontainer/containers/c1
```
Comment (duglin) [inline tests/sniffTest]: What's the issue with trying to stop and remove the container here? I'd prefer if each test cleaned-up after itself.

Comment (mikebrow) [inline tests/sniffTest]: There are timing issues with kill then delete... need to wait for the detached container to be in the destroyed state before killing it.  We could do the sleep 6, then delete c1 here if you prefer instead of the end of the script or use a smaller sleep in the tests then a smaller sleep after to wait for the windows to delete the detached containers.

Comment (cyphar) [inline tests/sniffTest]: Or could we do some sort of `while` loop while the state hasn't changed? I don't like magic sleep constants.

Comment (mikebrow) [inline tests/sniffTest]: while eval "runc state c1 | grep -q 'destroyed'" sleep...
then delete.. could do that

Comment (mikebrow) [inline tests/sniffTest]: Ok.. refreshed the PR to have it sleep/wait till the container is in the destroyed state.  Also cleaned up the output a bit by removing the trace output.. and fixed the alignment of the -e vs +e option.

File: tests/sniffTest [hunk 2]
```diff
@@ -47,9 +46,10 @@ echo "PASS: runc start"
 
 # Test exec
 ################################################################
-set -ex
+set -e 
 rm config.json && $RUNC spec  # reset things
 sed -i 's/"sh"/"sleep","6"/' config.json
+set +e
```
Comment (duglin) [inline tests/sniffTest]: why is this here?

Comment (mikebrow) [inline tests/sniffTest]: So that if an error occurs the sniff test continues.

**Comment by mrunalp** [comment]
Closing since we removed the sniff tests.
